app/models/concerns/user_omniauth_concern.rb

```diff
@@ -0,0 +1,52 @@
+module UserOmniauthConcern
+  extend ActiveSupport::Concern
+
+  included do
+    devise :omniauthable, omniauth_providers: [:facebook]
+  end
+
+  module ClassMethods
+    # Find or create a user from omniauth data.
+    #
+    # @param [Hash] auth The data with omniauth info.
+    # @return [User] The user found or created from omniauth data.
+    def from_omniauth(auth)
+      identity = User::Identity.find_by(provider: auth.provider, uid: auth.uid)
+      return identity.user if identity
+
+      User.create do |user|
+        user.update_from_general_auth_data(auth)
+        user.instance_users.build
+      end
+    end
+
+    # A convenience method that receives both parameters and session to initialize a user.
+    # This method is a override of devise {#new_with_session}
+    def new_with_session(params, session)
+      super.tap do |user|
+        facebook_data = session['devise.facebook_data']
```

Comment:
What's the correct way of declaring a private method in concern ? (don't need to include by user)